src/Sentry/SentrySdk.cs

```diff
@@ -52,17 +48,40 @@ internal static IHub InitHub(SentryOptions options)
         }
 
         // Initialize native platform SDKs here
-#if __MOBILE__
         if (options.InitNativeSdks)
         {
 #if __IOS__
             InitSentryCocoaSdk(options);
 #elif ANDROID
             InitSentryAndroidSdk(options);
+#elif NET8_0_OR_GREATER
+            if (AotHelper.IsNativeAot)
+            {
+                InitNativeSdk(options);
+            }
 #endif
         }
-#endif
-        return new Hub(options);
+
+        // We init the hub after native SDK in case the native init needs to adapt some options.
+        var hub = new Hub(options);
+
+        // Do the initial scope sync, if configured by the native integration.
+        if (options.NativeContextWriter is { } contextWriter)
+        {
+            hub.ConfigureScope((scope) =>
+            {
+                Task.Run(() => contextWriter.Write(scope)).ContinueWith(t =>
```

Comment:
This time worth adding some context so we don't scratch our heads why we did Task.Run here:


```suggestion
                // Write context asynchronously to reduce overhead on `Init`. 
                // Any exception is logged to avoid UnobservedTaskException
                Task.Run(() => contextWriter.Write(scope)).ContinueWith(t =>
```